src/starknet-docker-proxy.ts

```diff
@@ -32,7 +32,17 @@ export class StarknetDockerProxy extends DockerServer {
             volumes.push("-v", `${mirroredPath}:${mirroredPath}`);
         }
 
-        return [...volumes, "--network", "host"];
+        const dockerArgs = [...volumes];
+        // Check host machine
+        const isArm64 = process.arch === "arm64";
```

Comment:
This is an os problem, not an architecture one, the check for mac should be done as in `adaptUrl`

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/210/commits/73d4039c6eca8654e2f5d2d306b506f9d213cf1b